Scheduler/HPCSchedulerBasics/HPCSchedulerBasics/Program.cs

```diff
@@ -17,33 +17,88 @@ class Program
         static ISchedulerTask task;
         static ManualResetEvent jobFinishedEvent = new ManualResetEvent(false);
 
-        static void Main(string[] args)
+        static void ShowHelp()
+        {
+            string help = @"
+Usage:
+{0} [-c <cluster name>] [-u <user name>] [-d]
+
+Options:
+-c Provide a HPC cluster name to connect to. The default value is %CCP_SCHEDULER%.
+-u Provide a user name to connect as that user.
+-d Run in debug mode.
+";
+            Console.WriteLine(String.Format(help, System.Diagnostics.Process.GetCurrentProcess().ProcessName));
+        }
+
+        static int Main(string[] args)
         {
+            bool debug = false;
             string clusterName = Environment.GetEnvironmentVariable("CCP_SCHEDULER");
+            string userName = null;
+
+            for (int i = 0; i < args.Length; i++)
+            {
+                switch (args[i])
+                {
+                    case "-d":
+                        debug = true;
+                        break;
+                    case "-u":
+                        if (++i == args.Length)
+                        {
+                            ShowHelp();
+                            return 1;
+                        }
+                        userName = args[i];
+                        break;
+                    case "-c":
+                        if (++i == args.Length)
+                        {
+                            ShowHelp();
+                            return 1;
+                        }
+                        clusterName = args[i];
+                        break;
+                    default:
+                        ShowHelp();
+                        return 1;
+                }
+            }
 
-            if (args.Length > 0)
-            {
-                clusterName = args[0];
+            if (debug)
+            {
+                Console.Write("Press any key to start...");
+                Console.Read();
             }
 
             // Create a scheduler object to be used to 
             // establish a connection to the scheduler on the headnode
             using (IScheduler scheduler = new Scheduler())
             {
-                // Connect to the scheduler
-                Console.WriteLine("Connecting to {0}...", clusterName);
-                try
-                {
-                    scheduler.Connect(clusterName);
+                try
+                {
+                    if (userName != null)
+                    {
+                        // Connect to the scheduler as another user
+                        Console.WriteLine("Connecting to {0} as {1}...", clusterName, userName);
+                        scheduler.ConnectServiceAsClient(clusterName, () => userName);

```

Comment:
I believe `ConnectServiceAsClient` is not a public API which should be included in samples. Is there particular reason to include it?

Reply:
Oh, really? I thought it is since it's in the public interface of the SDK. I'm testing the "connect as client" functionality since a customer requires a similar one in REST API which I'm going to provide.